File: backend/model/diet.go
package model

import "time"

type DietRecord struct {
	ID          string     `json:"id"`
	ChildID     string     `json:"child_id"`
	FoodName    string     `json:"food_name"`
	FoodType    string     `json:"food_type"`
	AmountLevel int        `json:"amount_level"`
	RecordTime  time.Time  `json:"record_time"`
	MealGroupID *string    `json:"meal_group_id"`
	MealType    string     `json:"meal_type"`
	Notes       *string    `json:"notes"`
	CreatedBy   string     `json:"created_by"`
	CreatedAt   time.Time  `json:"created_at"`
}

File: web/src/api/diet.ts
// web/src/api/diet.ts
import api from './axios'

export interface DietRecord {
  id: string
  child_id: string
  food_name: string
  food_type: string
  amount_level: number
  record_time: string
  meal_group_id: string | null
  meal_type: string
  notes: string | null
  created_by: string
  created_at: string
}

export interface FoodType {
  value: string
  label: string
}

export const listDiet = (childId: string) =>
  api.get<DietRecord[]>(`/api/children/${childId}/diet`).then(r => r.data)

export const createDiet = (
  childId: string,
  data: {
    food_name: string
    food_type: string
    amount_level: number
    record_time: string
    meal_group_id?: string | null
    meal_type?: string
    notes?: string | null
  },
) =>
  api.post<DietRecord>(`/api/children/${childId}/diet`, data).then(r => r.data)

export const updateDiet = (
  childId: string,
  did: string,
  data: {
    food_name: string
    food_type: string
    amount_level: number
    record_time: string
    meal_group_id?: string | null
    meal_type?: string
    notes?: string | null
  },
) =>
  api.put<DietRecord>(`/api/children/${childId}/diet/${did}`, data).then(r => r.data)

export const deleteDiet = (childId: string, did: string) =>
  api.delete(`/api/children/${childId}/diet/${did}`)

export const getFoodTypes = (childId: string) =>
  api.get<FoodType[]>(`/api/children/${childId}/diet/types`).then(r => r.data)
